blog/views.py

```diff
@@ -0,0 +1,58 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.views import generic
+from django.views.generic.edit import FormMixin
+from .forms import CommentaryForm
+
+from blog.models import Post
+
+
+class IndexView(generic.ListView):
+    model = Post
+    posts = Post.objects.all().order_by("-created_time")
+    paginate_by = 5
+    template_name = "blog/index.html"
+    queryset = Post.objects.all().prefetch_related("commentary")
+
+
+class PostList(generic.ListView):
+    model = Post
+    template_name = "blog/post_list.html"
+    context_object_name = "post_list"
+    queryset = Post.objects.all().prefetch_related(
+        "commentary__user", "commentary__user__posts"
+    )
+    paginate_by = 5
+
+
+class PostDetail(generic.DetailView, FormMixin):
+    model = Post
+    template_name = "blog/post_detail.html"
+    context_object_name = "post"
+    form_class = CommentaryForm
+
+    def get_success_url(self):
+        return reverse("blog:post-detail", kwargs={"pk": self.object.pk})
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if "form" not in context:
+            context["form"] = self.get_form()
+        return context
+
+    @login_required
```

Comment:
The '@login_required' decorator will not work as expected here because it is applied to a class method. Consider using the 'LoginRequiredMixin' from 'django.contrib.auth.mixins' instead for class-based views.